feather/server/src/systems/player_join.rs

```diff
@@ -26,12 +31,114 @@ fn poll_new_players(game: &mut Game, server: &mut Server) -> SysResult {
 
 fn accept_new_player(game: &mut Game, server: &mut Server, client_id: ClientId) -> SysResult {
     let client = server.clients.get(client_id).unwrap();
-    client.send_join_game(server.options.default_gamemode);
+
+    let player_data = game.world.load_player_data(client.uuid());
+    let gamemode = player_data
+        .as_ref()
+        .map(|data| Gamemode::from_id(data.gamemode as u8).expect("Unsupported gamemode"))
+        .unwrap_or(server.options.default_gamemode);
+    let previous_gamemode = player_data
+        .as_ref()
+        .map(|data| PreviousGamemode::from_id(data.previous_gamemode as i8))
+        .unwrap_or(PreviousGamemode(None));
+
+    client.send_join_game(gamemode, previous_gamemode);
     client.send_brand();
 
-    let mut builder = game.create_entity_builder(Position::default(), EntityInit::Player);
+    // Abilities
+    let walk_speed = player_data
```

Comment:
I think it would be better to move this code to a separate function, in order not to clutter this one.
Maybe `fn player_data_or_default(data: Option<PlayerAbilities>, gamemode: Gamemode) -> PlayerAbilities`, with `Client::send_abilities` taking a `PlayerAbilities` struct instead.